### PR TITLE
Add bulk properties update

### PR DESCRIPTION
**Pull request by javierg**
This PR adds a method to update properties in bulk, it will set the same value to all the documents in the collection.

Open against 1.0 as I needed on some projects which still use this version of dolly

### CODE REVIEW

File: lib/dolly/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module Dolly
-  VERSION = "1.0.0"
+  VERSION = "1.0.1"
```
Comment (seancookr) [inline lib/dolly/version.rb]: didn't you already release `1.0.1`?

File: lib/dolly/collection.rb [hunk 2]
```diff
@@ -20,6 +20,15 @@ def last
       to_a.last
     end
 
+    def update_properties! properties ={}
+      properties.each do |key, value|
+        json.gsub! %r{\"#{key}\":\"[^\"]+\"}, "\"#{key}\":\"#{value}\""
```
Comment (seancookr) [inline lib/dolly/collection.rb]: can you explain this regex to me a little more. Im not sure what it does

File: test/collection_test.rb [hunk 3]
```diff
@@ -37,4 +37,9 @@ def setup
     assert_equal ["Foo B", "Foo A"], @collection.map(&:foo)
   end
 
+  test 'update attributes will change expected attributes' do
+    @collection.update_properties! foo: 'Foo 4 All'
+    assert_equal ['Foo 4 All', 'Foo 4 All'], @collection.map(&:foo)
```
Comment (seancookr) [inline test/collection_test.rb]: nice test!

File: lib/dolly/version.rb (hunk 1 above)
Comment (javierg) [inline lib/dolly/version.rb]: Yeah, butdont know why the branch doesnt have the version pushed

File: lib/dolly/collection.rb (hunk 2 above)
Comment (javierg) [inline lib/dolly/collection.rb]: json is the unparsed json string, so instead of parsing then locking for key and values to replace... I use regex, as it will be much faster

on the regex itself:

`\"` is a literral `\"` as the json string have the `"` escaped then I am matching the key of the property, and find the closing `\"` then there is a literal `:` following by the value enclosed in `\"`. The regex finds a `\"` followed by anything not that is what  `[^\"]` does, and we need to be something, that is why `+`, and now that I see this... I think this will fail if the value is empty or `nil`,

I will need to write more tests on this to ensure that the nil and empty values get updated.

**Comment by javierg** [comment]
@seancookr thanks to your inquires I discovered some failing edge cases, added some tests and fixes on the code 👍

**Comment by seancookr** [comment]
:shipit:

File: test/collection_test.rb [hunk 4]
```diff
@@ -37,4 +37,23 @@ def setup
...
+  test 'update empty attributes' do
+    json = '{"total_rows":2,"offset":0,"rows":[{"id":"foo_bar/0","key":"foo_bar","value":1,"doc":{"_id":"foo_bar/0","_rev":"7f66379ac92eb6dfafa50c94bd795122","foo":null,"bar":"","type":"foo_bar"}},{"id":"foo_bar/1","key":"foo_bar","value":1,"doc":{"_id":"foo_bar/1","_rev":"4d33cea0e55142c9ecc6a81600095469","foo":"Foo A","bar":"Bar A","type":"foo_bar"}}]}'
+    collection = Dolly::Collection.new json, FooBar
+
+    collection.update_properties! foo: 'Foo 4 All', bar: 'stuff'
+    assert_equal ['Foo 4 All', 'Foo 4 All'], collection.map(&:foo)
+    assert_equal ['stuff', 'stuff'], collection.map(&:bar)
+  end
+
+  test 'ypdate attributes will raise exception if property is missing' do
```
Comment (netlogics) [inline test/collection_test.rb]: typo

Comment (javierg) [inline test/collection_test.rb]: sorry closed... can be fixed on the repo

Comment (netlogics) [inline test/collection_test.rb]: by editing the file directly in github?

Comment (javierg) [inline test/collection_test.rb]: yes
